pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -324,3 +327,22 @@ func AddAlertAdminConfigMigrations(mg *migrator.Migrator) {
 	mg.AddMigration("create_ngalert_configuration_table", migrator.NewAddTableMigration(adminConfiguration))
 	mg.AddMigration("add index in ngalert_configuration on org_id column", migrator.NewAddIndexMigration(adminConfiguration, adminConfiguration.Indices[0]))
 }
+
+func AddProvisioningMigrations(mg *migrator.Migrator) {
+	provisioningTable := migrator.Table{
+		Name: "provenance_type",
+		Columns: []*migrator.Column{
+			{Name: "id", Type: migrator.DB_BigInt, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "org_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "record_key", Type: migrator.DB_NVarchar, Length: 190, Nullable: false},
+			{Name: "record_type", Type: migrator.DB_NVarchar, Length: 190, Nullable: false},
+			{Name: "provenance", Type: migrator.DB_NVarchar, Length: 190, Nullable: false},
```

Comment:
It's the exact same length we use for things like names, namespaces, etc, above in this file. For several of the record types that we want to manage, the name becomes the unique identifier, so synchronizing the same length for `record_key` made sense. There are a couple exceptions to this (alert rule UIDs being 40 characters), but 190 was a good common denominator that spans all our alerting types.

The others are really just for consistency. I normally prefer ints to back enum values instead of strings in databases, but I chose not to for `provenance` after we reached out to the core Grafana team. They have not yet fully settled on an enum for `provenance` and they want to keep it as a string, and rather just have objects "reserve" a few words. So, I stuck with the fixed length string, and used 190 again for consistency. `provenance` I think is a decent candidate to be shortened if we feel strongly about it, though.